core/templates/dev/head/filters.js

```diff
@@ -341,29 +341,39 @@ oppia.filter('normalizeWhitespacePunctuationAndCase', [function() {
   };
 }]);
 
-// Filter that targets to remove occurance of blank lines, empty paragraphs
+// Filter that targets to remove occurrence of blank lines, empty paragraphs
 // and spaces at the end of string. It'll remove <p></p>, <p><br></p>, spaces
-// and &nbsp; occurances from the end of the string.
+// and &nbsp; occurrences from the end of the string.
 oppia.filter('removeExtraLinesAndSpace', [function() {
   return function(string) {
     if (!angular.isString(string)) {
       return string;
     }
     var BLANK_LINES_TEXT = '<p><br></p>';
     var EMPTY_PARA_TEXT = '<p></p>';
+    var BLANK_CHAR = ' ';
+    var BREAK_LINE_TAG = '<br>';
+    var SPACE_ENTITY = '&nbsp;';
     while (1) {
-      var strlength = string.lastIndexOf('<');
-      if (string[strlength - 1] !== ' ' &&
-          string.substring(strlength - 4, strlength) !== '<br>' &&
-          string.substring(strlength - 6, strlength) !== '&nbsp;') {
+      var lastIndexOfClosingTag = string.lastIndexOf('<');
+      if (string[lastIndexOfClosingTag - 1] !== BLANK_CHAR &&
+          string.substring(
+            lastIndexOfClosingTag - 4, lastIndexOfClosingTag
```

Comment:
There are still magic numbers lying around, this should be BREAK_LINE_TAG.length instead of 4, ditto elsewhere.

Reply:
Oh got it! Done changes

Reply:
Also would be better to explain in comments why -1 is used?